hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -143,31 +147,48 @@ else if ( explicitDialectConfiguration( explicitDatabaseName, configurationValue
 					explicitDatabaseMinorVersion,
 					explicitDatabaseVersion
 			);
+			databaseConnectionInfo = buildDbInfo(false, registry, configurationValues);
 		}
 		else {
 			jdbcEnvironment = getJdbcEnvironmentWithDefaults( configurationValues, registry, dialectFactory );
+			databaseConnectionInfo = buildDbInfo( false, registry, configurationValues);
 		}
 
-		logDbInfo( registry, jdbcEnvironment );
+		// most likely, the version hasn't been set yet, at least not for the ConnectionProviders that we currently maintain
+		databaseConnectionInfo.setDBVersion( jdbcEnvironment.getDialect().getVersion() );
+
+		// Standardized DB info logging
+		ConnectionInfoLogger.INSTANCE.logConnectionInfoDetails( databaseConnectionInfo.getDBInfoAsString() );
 
 		return jdbcEnvironment;
 	}
 
-	private static void logDbInfo(ServiceRegistryImplementor registry, JdbcEnvironment jdbcEnvironment) {
-		// Standardized DB info logging
-		DatabaseConnectionInfo databaseConnectionInfo = null;
-		if ( !isMultiTenancyEnabled( registry ) ) {
-			final ConnectionProvider cp = registry.requireService( ConnectionProvider.class );
-			databaseConnectionInfo = cp.getDatabaseConnectionInfo();
+	private DatabaseConnectionInfo buildDbInfo(boolean isJdbcMetadataAccessAllowed, ServiceRegistryImplementor registry, Map<String, Object> configurationValues) {
```

Comment:
I'd split these into separate methods.  Its minor, but we've already done this if-check in the enclosing callers.